api/api/src/main/scala/uk/ac/wellcome/platform/api/Router.scala

```diff
@@ -0,0 +1,246 @@
+
+class Router(elasticClient: ElasticClient,
+             elasticConfig: DisplayElasticConfig,
+             apiConfig: ApiConfig)(implicit ec: ExecutionContext)
+    extends FailFastCirceSupport
+    with Directives {
+
+  import MultipleWorksResponse.encoder
+  import ResultResponse.encoder
+
+  def routes: Route = handleRejections(rejectionHandler) {
+    ignoreTrailingSlash {
+      pathPrefix(apiConfig.pathPrefix) {
+        concat(
+          pathPrefix("v2") {
+            concat(
+              path("works") {
+                MultipleWorksParams.parse { params =>
+                  getWithFuture(multipleWorks(params))
+                }
+              },
+              path("works" / Segment) { workId =>
+                SingleWorkParams.parse { params =>
+                  getWithFuture(singleWork(workId, params))
+                }
+              },
+              path("context.json") {
+                getFromFile(context.getPath)
+              },
+              path("swagger.json") {
+                swagger
+              }
+            )
+          },
+          path("v1" / Remaining) { _ =>
+            v1ApiGone
+          },
+          pathPrefix("management") {
+            concat(
+              path("healthcheck") {
+                get {
+                  complete("message" -> "ok")
+                }
+              },
+              path("clusterhealth") {
+                getClusterHealth
+              }
+            )
+          }
+        )
+      }
+    }
+  }
+
+  def multipleWorks(params: MultipleWorksParams): Future[Route] = {
+    val searchOptions = params.searchOptions(apiConfig)
+    val index = params._index.map(Index(_)).getOrElse(elasticConfig.indexV2)
+    worksService
+      .listOrSearchWorks(index, searchOptions, params.workQuery)
+      .map {
+        case Left(err) => elasticError(err)
+        case Right(resultList) =>
+          extractUri { uri =>
+            contextUri { contextUri =>
+              complete(
+                MultipleWorksResponse(
+                  resultList,
+                  searchOptions,
+                  params.include.getOrElse(V2WorksIncludes()),
+                  uri,
+                  contextUri
+                )
+              )
+            }
+          }
+      }
+  }
+
+  def singleWork(id: String, params: SingleWorkParams): Future[Route] = {
+    val index = params._index.map(Index(_)).getOrElse(elasticConfig.indexV2)
+    val includes = params.include.getOrElse(V2WorksIncludes())
+    worksService
+      .findWorkById(id)(index)
+      .map {
+        case Right(Some(work: IdentifiedWork))           => workFound(work, includes)
+        case Right(Some(work: IdentifiedRedirectedWork)) => workRedirect(work)
+        case Right(Some(_))                              => workGone
+        case Right(None)                                 => workNotFound(id)
+        case Left(err)                                   => elasticError(err)
```

Comment:
Out of interest - do we think this is the only error that will pop up?

Reply:
No there are other errors that can be raised at different parts of the routing, but here we are matching on an `Either[ElasticError, Option[Work]]`, so it's the only error response we can produce a this point in the code.